numpy/core/src/npysort/quicksort.c.src

```diff
@@ -518,6 +526,8 @@ stack_pop:
     }
 
     free(vp);
+    if (PyErr_Occurred())
```

Comment:
I don't think that any of the previous `PyErr_Occurred() ` can ever be triggered. Certainly not for the numeric sorts, which could run with the GIL released as far as i can tell, extremely unlikely for the string sorts, which call `PyArray_ITEMSIZE` which is a struct access which is extremely unlikely to ever fail. Even more, since, `PyArray_Sort` releases the GIL when calling these functions if the dtype doesn't require it, that call to `PyErr_Occurred` without reacquiring the GIL is likely to crash the interpreter.